onnxruntime/core/providers/webnn/builders/impl/qdq_op_builder.cc

```diff
@@ -42,70 +42,28 @@ Status QDQOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
   int32_t zero_point_type = 0;
   ORT_RETURN_IF_NOT(GetType(*input_defs[0], input_type, logger), "Cannot get input data type");
   ORT_RETURN_IF_NOT(GetType(*output_defs[0], output_type, logger), "Cannot get output data type");
+
   emscripten::val input = model_builder.GetOperand(input_defs[0]->Name());
   emscripten::val scale = model_builder.GetOperand(input_defs[1]->Name());
-
   emscripten::val zero_point = emscripten::val::null();
+
   if (input_defs.size() == 3 && input_defs[2]->Exists()) {
     zero_point = model_builder.GetOperand(node.InputDefs()[2]->Name());
   } else {
     // DequantizeLinear: x_zero_point's data type equals to input data type
     // QuantizeLinear: x_zero_point's data type equals to output data type
+    // WebNN requires the zero_point to have the same shape as the scale
     zero_point_type = op_type == "DequantizeLinear" ? input_type : output_type;
-    zero_point = model_builder.GetZeroConstant(zero_point_type);
-  }
-
-  emscripten::val output;
-  NodeAttrHelper helper(node);
-  int32_t axis = helper.Get("axis", 1);
```

Comment:
Is this deleted `axis` code not still useful? If `block_size` is not specified and `axis` is present, we can reshape the scale and zp to be rank compatible:

e.g.
```
input rank = 4
axis = 1
```
```
if block size present
    pass through sizes as-is
else if axis present
    int32_t axis = helper.Get("axis", 1);
    scale = ProjectToRank(model_builder, scale, input.rank, axis);
    zero_point = ProjectToRank(model_builder, scale, input.rank, axis);
endif

// ReshapeToRank is a helper that takes an existing tensor, new target rank, and target dimensions to
// reshape the existing dimensions to. e.g. Given 2D input of sizes [5,6], target rank of 4, and target axes
// of [1,3], the new shape would be [1,5,1,6].
```

Reply:
Oh, you are right, `axis` should be used for broadcasting.

But one block here is current `reshape` doesn't support int4/uint4 data type. We should support this in Chromium.

cc/ @lisa0314

Reply:
> But one block here is current reshape doesn't support int4/uint4 data type.

Good point - most ops do not need to support int4 (nor do we want them to), but at least `reshape` should!